### PR TITLE
scaffolding for saving log tarball on failures

### PR DESCRIPTION
**Pull request by joelddiaz**
allow storing the ssh private key in the ssh secret

extend installmanager to set up ssh agent (if necessary), read in terraform file, generate script to fetch logs, and run the script to fetch the logs

TODO: store the saved log somewhere (S3, the in-cluster registry)

### CODE REVIEW

File: contrib/pkg/installmanager/installmanager_test.go [hunk 1]
```diff
@@ -454,3 +455,253 @@ last line with password in text`,
...
+						}
+					}
+				}
+			}
+		}
+	]
+}`, instanceName, ip)
+}
+
+func TestInstallManagerSSH(t *testing.T) {
```
Comment (csrwng) [inline contrib/pkg/installmanager/installmanager_test.go]: @joelddiaz I think we could simplify this test to limit it to testing your code (which basically parses ssh-agent output, sets env vars and calls ssh-add). You could supply alternate binaries for `ssh-agent` and `ssh-add` just to ensure that they are properly being called (and that you're properly parsing the output)

Comment (csrwng) [reply contrib/pkg/installmanager/installmanager_test.go]: we kind of already do that with the installer binary: https://github.com/openshift/hive/blob/34814cafa464c8040918a15767d758ce1a957c9a/contrib/pkg/installmanager/installmanager_test.go#L55-L64

File: config/templates/cluster-deployment.yaml [hunk 2]
```diff
@@ -12,6 +12,10 @@ parameters:
   displayName: SSH Key
   description: Your public SSH key to reach instances.
   required: true
+- name: SSH_PRIV_KEY
```
[the remark this answers is not in the document]
Comment (joelddiaz) [reply config/templates/cluster-deployment.yaml]: k. also updated create-cluster with updated PR

File: contrib/pkg/installmanager/installmanager_test.go (hunk 1 above)
Comment (joelddiaz) [reply contrib/pkg/installmanager/installmanager_test.go]: k. have some fake ssh-agent and ssh-add instead of relying on actual binaries

**Comment by csrwng** [comment]
@joelddiaz you have a mismatch between the arg name declared in installmanager and the one passed in the generated install job. 
/lgtm cancel

**Comment by joelddiaz** [comment]
for posterity, if we decide to try to upload logs to a registry, something like this could be a starting point

```
package main

import (
        "context"
        "fmt"

        "github.com/containers/image/copy"
        "github.com/containers/image/signature"
        "github.com/containers/image/transports/alltransports"
)

func main() {
        var ctx context.Context
        ctx = context.TODO()

        var policy *signature.Policy
        policy, err := signature.DefaultPolicy(nil)
        if err != nil {
                panic("error creating default policy")
        }
        policyCtx, err := signature.NewPolicyContext(policy)
        if err != nil {
                panic("error creating policy context")
        }
        srcRef, err := alltransports.ParseImageName("tarball://home/jdiaz/log-bundle.tar.gz")
        if err != nil {
                panic("error setting source ref")
        }
        destRef, err := alltransports.ParseImageName("docker://quay.io/jdiaz/testing123")
        if err != nil {
                panic("error setting destination ref")
        }
        result, err := copy.Image(ctx, policyCtx, destRef, srcRef, &copy.Options{})
        if err != nil {
                fmt.Printf("ERR: %+v\n", err)
                panic("error copying image")
        }
        fmt.Printf("RES: %+v\n", string(result))
}

```

File: pkg/install/generate.go [hunk 3]
```diff
@@ -223,7 +252,8 @@ func GenerateInstallerJob(
 				// a sleep-seconds.txt file. If one is written, we will sleep that number of seconds. This allows exponential backoff
 				// for failing installs.
 				fmt.Sprintf(
-					"/usr/bin/hiveutil install-manager --work-dir /output --log-level debug --install-config /installconfig/install-config.yaml --region %s %s %s; installer_result=$?; if [ -f /output/sleep-seconds.txt ]; then sleep_seconds=$(cat /output/sleep-seconds.txt); echo \"sleeping for $sleep_seconds seconds until next retry\"; sleep $sleep_seconds; fi; exit $installer_result",
+					"/usr/bin/hiveutil install-manager --work-dir /output --log-level debug --install-config /installconfig/install-config.yaml --ssh-priv-key-path %s --region %s %s %s; installer_result=$?; if [ -f /output/sleep-seconds.txt ]; then sleep_seconds=$(cat /output/sleep-seconds.txt); echo \"sleeping for $sleep_seconds seconds until next retry\"; sleep $sleep_seconds; fi; exit $installer_result",
```
Comment (dgoodwin) [inline pkg/install/generate.go]: Old versions of the install image (i.e. all current cluster image sets) do not have this new flag, and thus all installs are now broken in stage. I don't think this has made it to prod.

**Pull request merged**